src/main/java/edu/ucla/library/services/metadata/MetadataSetter.java

```diff
@@ -0,0 +1,350 @@
+            return PROBE_DOESNT_EXIST;
+        }
+
+        try {
+            final Path basePath = FileSystems.getDefault().getPath(CSV_PATH);
+            if (Files.isDirectory(basePath)) {
+                Files.find(Paths.get(CSV_PATH), Integer.MAX_VALUE,
+                    (filePath, fileAttr) -> fileAttr.isRegularFile()
+                     && filePath.toFile()
+                                .getName()
+                                .endsWith("csv"))
+                    .forEach(path -> addMetaToCsv(path));
+            } else if (Files.isRegularFile(basePath)) {
+                addMetaToCsv(basePath);
+            }
+        } catch (IOException details) {
+            System.err.println("Problem reading file/walking file directory: "
+                         + details.getMessage());
+        }
+        return 0;
+    }
+
+  /**
+   * Verify a directory/file exists.
+   *
+   * @param aFileName pathed name of file to test
+   * @return true/false for file/directory existence.
+  */
+    public static boolean fileDirExists(final String aFileName) {
+        if (!Files.exists(FileSystems.getDefault().getPath(aFileName))) {
+            System.err.println("Directory/file must exist: " + aFileName);
+            return false;
+        }
+        return true;
+    }
+
+  /**
+   * Verify ffprobe executable exists.
+   *
+   * @param aFileName pathed name of executable to test
+   * @return true/false for valid ffprobe executable.
+  */
+    public static boolean validFFProbe(final String aFileName) {
+        try {
+            new FFprobe(FFMPEG_PATH).version();
+        } catch (IOException e) {
+            System.err.println(FFMPEG_PATH + " is not valid path to ffprobe");
+            return false;
+        }
+        return true;
+    }
+
+  /**
+   * Method to process CSV file(s) to add columns for media metadata.
+   *
+   * @param aPath Path to file to be read and copied/updated
+  */
+    private static void addMetaToCsv(final Path aPath) {
+        final CSVReader reader;
+        final CSVWriter writer;
+        final List<String[]> input;
+        final List<String[]> output;
+        final boolean hasAllMetas;
+
+        System.out.println("working with file " + aPath); //.toFile().getName());
+        try {
+            reader = new CSVReader(new FileReader(aPath.toFile()));
+            input = reader.readAll();
+            reader.close();
+
+            output = new ArrayList<>(input.size());
+            hasAllMetas = allMetaFieldsPresent(input.get(0));
+            if (!hasAllMetas) {
+                output.add(buildHeaderRow(input.get(0)));
+            } else {
+                output.add(input.get(0));
+            }
+
+            for (int index = 1; index < input.size(); index++) {
+                output.add(buildARow(hasAllMetas, input.get(index)));
+            }
+
+            writer = new CSVWriter(new FileWriter(new File(
+               OUTPUT_PATH.concat(aPath.toFile().getName()))));
```

Comment:
I think it would be ideal if we created the output directory if it doesn't exist, but I'm not that opinionated on it.